test/testsuites/csctargetnamespacetests.go

```diff
@@ -82,66 +76,49 @@ func CscTargetNamespace(t *testing.T) {
 // nonexistent namespace.
 func configuringStateWhenTargetNamespaceDoesNotExist(t *testing.T) {
 	namespace, err := test.NewTestCtx(t).GetNamespace()
-	if err != nil {
-		t.Fatalf("Could not get namespace: %v", err)
-	}
+	require.NoError(t, err, "Could not get namespace")
 
 	// Check that the CatalogSourceConfig with an non-existing targetNamespace eventually reaches the
 	// configuring phase with the expected message.
 	expectedPhase := "Configuring"
 	expectedMessage := fmt.Sprintf("namespaces \"%s\" not found", targetNamespace)
 	err = helpers.WaitForExpectedPhaseAndMessage(test.Global.Client, cscName, namespace, expectedPhase, expectedMessage)
-	if err != nil {
-		t.Fatalf("CatalogSourceConfig never reached expected phase/message, expected %v/%v", expectedPhase, expectedMessage)
-	}
+	assert.NoError(t, err)
 }
 
 // succeededStateAfterTargetNamespaceCreated is a test case that confirms that a csc that had a
 // targetNamespace which did not exist eventually reaches a succeeded state once the targetNamespace is created.
 func succeededStateAfterTargetNamespaceCreated(t *testing.T) {
 	// Get test namespace.
 	namespace, err := test.NewTestCtx(t).GetNamespace()
-	if err != nil {
-		t.Fatalf("Could not get namespace: %v", err)
-	}
+	require.NoError(t, err, "Could not get namespace")
 
 	// Now that the targetNamespace has been created, periodically check that the CatalogSourceConfig
 	// has reached the Succeeded phase.
 	expectedPhase := "Succeeded"
 	err = helpers.WaitForExpectedPhaseAndMessage(test.Global.Client, cscName, namespace, expectedPhase, "")
-	if err != nil {
-		t.Fatalf("CatalogSourceConfig never reached expected phase, expected %v", expectedPhase)
-	}
+	assert.NoError(t, err)
 }
 
 // childResourcesCreated checks that once a CatalogSourceConfig is created that all expected runtime
 // objects are created as well.
 func childResourcesCreated(t *testing.T) {
 	// Get test namespace.
 	namespace, err := test.NewTestCtx(t).GetNamespace()
-	if err != nil {
-		t.Fatalf("Could not get namespace: %v", err)
-	}
+	require.NoError(t, err, "Could not get namespace")
 	// Check that the CatalogSourceConfig and its child resources were created.
 	err = helpers.CheckCatalogSourceConfigAndChildResourcesCreated(test.Global.Client, cscName, namespace, targetNamespace)
-	if err != nil {
-		t.Fatal(err)
-
-	}
+	assert.NoError(t, err)
```

Comment:
My suggestion is for this helper to return the error and have `asserts` or `require` in the call site. This comment applies for all helpers. 
PS: Please wait for comment from others before making this change.

Reply:
@aravindhp looks like helpers like these are just returning the `client`'s result. Doesn't look like the helpers need to return custom errors, and can just return whatever error the `client` might return.

Reply:
My point was to not have `asserts` or `require` in the helpers. I never said anything about custom errors.

Reply:
+1, there's no reason for the helpers to have anything succeed or fail directly. They should just return information about what occurred, and the test methods should do the `asserting` and `requiring`

Reply:
@aravindhp `csctargetnamespacetests.go` is a test file. None of the helper functions use `asserts` or `require` as of now. They return the error, which are then `assert`-ed or `require`-ed at the call site, as you were suggesting.

Reply:
`childResourcesCreated()` for practical purposes is a helper function. Point I have been trying to get across is to call `asserts` and `requires` in the main test functions and nowhere else.

Reply:
@aravindhp looks like `childResourcesCreated()` is a func that tests an individual unit (existence of child resources after creation of a CSC). It uses a helper itself, i.e `CheckCatalogSourceConfigAndChildResourcesCreated()`, and it itself isn't being called by any other function as a `helper`. I don't see how `childResourcesCreated()` is a `helper` function.